### DISCUSSION TITLE
MPNowPlayingInfoCenter don't pause when playbackRate is set to 0

### DISCUSSION THREAD

**Discussion by XZB-1248** [sbooth/SFBAudioEngine#344] [Show and tell]
I encountered an issue: when I set `MPNowPlayingInfoCenter`'s `MPNowPlayingInfoPropertyPlaybackRate` to 0, the lock screen still shows the play button, but the progress no longer increases.
Eventually, I found out that when calling `player.pause`, the audioEngine continued to push silences (the note for `playerNode.pause` is "Pauses audio from the current decoder and pushes silence"), which led the system to think that audio was still playing, thus it continued to display the pause button.

This problem troubled me for several days, and after multiple tests, I found that the issue could be resolved by calling `player.playerNode.engine?.pause` after `player.pause`, and `player.playerNode.engine?.start` before `player.resume` or `player.play`.

Although it’s not recommended to directly manipulate `AVAudioEngine`, it indeed solved my problem.
I hope this can help others.

**Comment by sbooth** [comment]
It might be worth adding an optional argument to `pause` that would allow pausing the `AVAudioEngine` as well. The reason behind the way things work now is that pausing `AVAudioEngine` also stops the hardware, which on iOS could make sense, but with external devices on macOS isn't always ideal.
